src/lib/di/runtime_provider.hpp

```diff
@@ -0,0 +1,183 @@
+//
+// Copyright (c) 2012-2020 Kris Jusiak (kris at jusiak dot net)
+//
+// Distributed under the Boost Software License, Version 1.0.
+// (See accompanying file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+#pragma once
+
+#ifndef CANARY_RUNTIME_PROVIDER_HPP
+	#define CANARY_RUNTIME_PROVIDER_HPP
+
+	#include <cassert>
+	#include <functional>
+	#include <memory>
+	#include <type_traits>
+	#include <typeindex>
+	#include <unordered_map>
+	#include <utility>
```

Comment:
Is it really necessary? It's already on pch.hpp too

Reply:
No it's not, at least I didn't find any of those in pch.